java/integTest/src/test/java/glide/BaseClientTests.java

```diff
@@ -0,0 +1,416 @@
+                            .build();
+            client.set("set_expired_value", INITIAL_VALUE, options).get(10, SECONDS);
+            var data = client.get("set_expired_value").get(10, SECONDS);
+            assertNull(data);
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void set_missing_value_and_returnOldValue_is_null() {
+        for (BaseClient client : clients) {
+            String ok = client.set(KEY_NAME, INITIAL_VALUE).get(10, SECONDS);
+            assertEquals(OK, ok);
+
+            var options = SetOptions.builder().returnOldValue(true).build();
+            var data = client.set("another", ANOTHER_VALUE, options).get(10, SECONDS);
+            assertNull(data);
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void decr_and_decrBy_existing_key() {
+        String key = RandomStringUtils.randomAlphabetic(10);
+
+        for (BaseClient client : clients) {
+            assertEquals(OK, client.set(key, "10").get(10, SECONDS));
+
+            assertEquals(client.decr(key).get(10, SECONDS), 9);
+            assertEquals(client.get(key).get(10, SECONDS), "9");
+
+            assertEquals(client.decrBy(key, 4).get(10, SECONDS), 5);
+            assertEquals(client.get(key).get(10, SECONDS), "5");
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void decr_and_decrBy_non_existing_key() {
+        String key1 = RandomStringUtils.randomAlphabetic(10);
+        String key2 = RandomStringUtils.randomAlphabetic(10);
+
+        for (BaseClient client : clients) {
+            assertNull(client.get(key1).get(10, SECONDS));
+            assertEquals(client.decr(key1).get(10, SECONDS), -1);
+            assertEquals(client.get(key1).get(10, SECONDS), "-1");
+
+            assertNull(client.get(key2).get(10, SECONDS));
+            assertEquals(client.decrBy(key2, 3).get(10, SECONDS), -3);
+            assertEquals(client.get(key2).get(10, SECONDS), "-3");
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void incr_commands_existing_key() {
+        String key = RandomStringUtils.randomAlphabetic(10);
+
+        for (BaseClient client : clients) {
+            assertEquals(OK, client.set(key, "10").get(10, SECONDS));
+
+            assertEquals(client.incr(key).get(10, SECONDS), 11);
+            assertEquals(client.get(key).get(10, SECONDS), "11");
+
+            assertEquals(client.incrBy(key, 4).get(10, SECONDS), 15);
+            assertEquals(client.get(key).get(10, SECONDS), "15");
+
+            assertEquals(client.incrByFloat(key, 5.5).get(10, SECONDS), 20.5);
+            assertEquals(client.get(key).get(10, SECONDS), "20.5");
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void incr_commands_non_existing_key() {
+        String key1 = RandomStringUtils.randomAlphabetic(10);
+        String key2 = RandomStringUtils.randomAlphabetic(10);
+        String key3 = RandomStringUtils.randomAlphabetic(10);
```

Comment:
you randomize the key, but not the increment value?

Reply:
Took these from the Python and node IT tests.